### DISCUSSION TITLE
How to trigger a GitHub action only when the response of a script is different and it changes a particular file

### DISCUSSION THREAD

**Discussion by idontknowjs** [community/community#24919] [Actions]
<p><em>Use case:</em> I’m having a python script to fetch a ICS (iCalendar) feed, which parses the ICS into a YAML file.</p>
<p>How can I create a GitHub action, which will schedule the script to run every day and check whether after it’s execution, the YAML files changes or not.</p>
<p>If the YAML file changes, then only it might send a PR</p>

**Comment by airtower-luna** [comment]
<p>You can use a <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events">scheduled event</a> to trigger the workflow. As for detecting changes I’d suggest just overwriting the YAML file and then using <a href="https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code"><code>git diff --exit-code</code></a> to detect if it actually changed.</p>

**Comment by idontknowjs** [comment]
<p>Hey, thanks for your reply!</p>
<p>Do you mean something like this with a scheduled event:</p>
<pre><code class="lang-auto">       # Runs the Script to convert ics to yaml format

      - name: Run ics-to-yml.py
        run: python assets/script/ics-to-yml.py

      - name: Check git diff
        if: git diff --exit-code == 1
        # Creates Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
          token: ${{ secrets.PAT }}
          branch-suffix: timestamp
          path: ./master
          title: Update upcoming events
          labels: automated-build
          commit-message: Update upcoming events
</code></pre>
<p>Is there any other way to do this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Close, but you can’t use a shell command directly in the <code>if</code> condition. You’ll have to use an extra step to set an output or environment variable, e.g.:</p>
<pre><code class="lang-yaml">- name: Check diff
  run: |
    if git diff --exit-code; then
        echo "CHANGED=false" &gt;&gt;${GITHUB_ENV}
    else
        echo "CHANGED=true" &gt;&gt;${GITHUB_ENV}
    fi

- name: Create pull request
  if: env.CHANGED == 'true'
  # ...
</code></pre>

**Comment by idontknowjs** [comment]
<p>I really appreciate for your help, thank youu!</p>

**Comment by idontknowjs** [comment]
<p>Hey, I’m not sure why it’s saying it’s not a git repository. Thus all the jobs run successfully, but no changes occur</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/e/0ec3fe7a4f801beebe6aa5c8e545dbaecd893c37.png" data-download-href="/uploads/short-url/26CBc3kXriGHe9sf3Y6yjluyvQP.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/53327173/181087750-b7113b54-1fa9-46d8-bafd-7e5e6f26a6df.png" alt="image" data-base62-sha1="26CBc3kXriGHe9sf3Y6yjluyvQP" width="690" height="284" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/e/0ec3fe7a4f801beebe6aa5c8e545dbaecd893c37_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">957×394 11 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>Did you use <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> before the other things? If no, you definitely need to do that.</p>
<p>If yes check its logs. By default it creates a Git repository, but if the installed Git version is too old (that might happen with a job container or self-hosted runner) it might just download the files over the API.</p>

**Comment by idontknowjs** [comment]
<p>Yes, I used actions/checkout before these.</p>
<p>I think in this, the git is initialized:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/c/2cf595ec8913c2aac5521a58445832c858f74abc.png" data-download-href="/uploads/short-url/6pJaKXMCdrxByddREimxLaKfkcs.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/53327173/181087751-40630422-cb13-425c-8e32-bd3a121f3af8.png" alt="image" data-base62-sha1="6pJaKXMCdrxByddREimxLaKfkcs" width="690" height="360" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/c/2cf595ec8913c2aac5521a58445832c858f74abc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">948×495 30.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>Looks like it is, but the working directory looks odd with the <code>/master</code> part at the end. Is that the same directory the file gets written to and that <code>git diff</code> is running in?</p>
